java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -181,11 +179,11 @@ void release(final BufferLedger ledger) {
   /**
    * Return the size of underlying chunk of memory managed by this Allocation Manager.
    *
+   * <p>The underlying memory chunk managed can be different from the original requested size.
+   *
    * @return size of underlying memory chunk
    */
-  public long getSize() {
-    return size;
-  }
+  protected abstract long getSize();
```

Comment:
Oh shoot, I just realized you're reducing visibility her on a public interface. You should keep this public. Sorry I missed this.

Reply:
oops. I was actually meaning to check in with you on this. (I think this interface should be protected) but sure we should keep it public for backward compatibility issue.